test/e2e/framework/util.go

```diff
@@ -1359,14 +1359,14 @@ func podNotPending(c clientset.Interface, podName, namespace string) wait.Condit
 // to terminate or if the pod terminated with an unexpected reason.
 func waitForPodTerminatedInNamespace(c clientset.Interface, podName, reason, namespace string) error {
 	return WaitForPodCondition(c, namespace, podName, "terminated due to deadline exceeded", PodStartTimeout, func(pod *v1.Pod) (bool, error) {
-		if pod.Status.Phase == v1.PodFailed {
-			if pod.Status.Reason == reason {
+		if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
+			if reason == "" ||
+				strings.ToLower(strings.TrimSpace(pod.Status.Reason)) == strings.ToLower(strings.TrimSpace(reason)) {
```

Comment:
Well, my concern is, since Reason is an optional string and there are only a couple of constants defined for a pod's Status' Reason, that a caller could mis-type the reason or follow other code that passes "" as the reason, and thus cause the loop to run longer. (In this scenario the loop would run until the pod's `Get` failed or 5m.) Yes, reason is being sanitized but not my much, and I wouldn't think the reason's character case or intermediate spaces would matter. However, I'm ok dropping this sanitation. (BTW, I found only one case of a non-blank reason, in _e2e/common/pods.go_, where "DeadlineExceeded" is passed, and this works the same with my change.)

For the case of "" passed as the reason. Does this mean that the caller wants to wait until the pod's Reason is blank (even if its Status is "Failed"), or does it mean that the reason doesn't matter? There should, IMO, be a way to say "I want to wait until this pod is deleted regardless of why". To me a blank reason seems like a good way to do that.